bson/struct_codec.go

```diff
@@ -216,14 +181,17 @@ func (sc *StructCodec) EncodeValue(ec EncodeContext, vw ValueWriter, val reflect
 		}
 	}
 
-	if sd.inlineMap >= 0 {
+	if sd.inlineMap >= 0 && sc.elemEncoder != nil {
```

Comment:
This check will result in omitting the map elements if there is no `elemEncoder`, which could result in a subtle bug that drops data. The only code that creates a `structCodec` is the `bson` package, so that's not a case we need to handle gracefully. A panic would be a much more obvious indicator of such a bug. We should remove this nil check.

```suggestion
	if sd.inlineMap >= 0 {
```

Additionally, we could panic in `newStructCodec` if `elemEncoder` is nil, although not all code paths that use a `structCodec` call `newStructCodec`.